tungsten_ci_utils/generate_build_change_info/last_successful.py

```diff
@@ -0,0 +1,95 @@
+import sys
+import argparse
+import MySQLdb
+import logging
+import json
+
+log = logging.getLogger(__name__)
+
+def set_logging():
+    log.setLevel(logging.DEBUG)
+    formatter = logging.Formatter(
+        '%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    console = logging.StreamHandler()
+    console.setLevel(logging.DEBUG)
+    console.setFormatter(formatter)
+    log.addHandler(console)
+
+set_logging()
+
+def get_json_data(file):
+    json_file = open(file).read()
+    data = json.loads(json_file)
+    return data
+
+def close_db_exit_err(db):
+    if db is not None:
+        db.close()
+    sys.exit(1)
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--credentials-json", action="append")
+    parser.add_argument("branch")
+    parser.add_argument("build_number")
+    args = parser.parse_args()
+    branch = str(args.branch)
+    credentials_json = args.credentials_json[0]
+
+    build_number = og_build_number = int(args.build_number)
+    build_number -= 1
+
+    db_config = get_json_data(credentials_json)
+
+    try:
+        db = MySQLdb.connect(
+            user=db_config["user"],
+            passwd=db_config["passwd"],
+            db=db_config["db"],
+            host=db_config["host"],
+            port=db_config["port"]
+        )
+        cur = db.cursor()
+        log.debug('connected to db')
+    except:
+        log.error('database connection error')
+        sys.exit(1)
+
+    while build_number > 0:
+        query = """
+        SELECT result FROM zuul_buildset WHERE id IN 
+        (SELECT buildset_id FROM zuul_build WHERE log_url LIKE CONCAT('%%periodic-nightly%%/', %s, '/%s%%'))
```

Comment:
Try matching branch and pipeline explicitly using SQL column names (will be more readable in the future, compared to matching the log URL string).

Reply:
Also, you can directly add a condition "and result == 'SUCCESS'" here, unless you wanted to iterate and log the failed builds.